src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,21 +1233,135 @@ private int SkipBlockComment(int i)
-                case 'e': return '\u001b';
-                case 'f': return '\f';
-                case 'n': return '\n';
-                case 'r': return '\r';
-                case 't': return '\t';
-                case 'v': return '\v';
-                default: return c;
+                case '0': return "\0";
+                case 'a': return "\a";
+                case 'b': return "\b";
+                case 'e': return "\u001b";
+                case 'f': return "\f";
+                case 'n': return "\n";
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
+            }
+        }
+
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            bool isBracketedSequence = false;
+            bool isTerminated = false;
+
+            char c = PeekChar();
+            if (c == '{')
+            {
+                SkipChar();
+                isBracketedSequence = true;
+            }
+
+            int maxNumberOfHexDigits = (isBracketedSequence ? 6 : 4);
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
+            {
+                c = GetChar();
+
+                // Bracketed sequence has been terminated.
+                if (isBracketedSequence && (c == '}'))
+                {
+                    if (i == 0)
+                    {
+                        // Variable sequence must have at least one hex char.
+                        IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                        ReportError(errorExtent, () => ParserStrings.EmptyUnicodeEscapeSequence);
+                        return s_unknownUnicodeChar;
+                    }
+
+                    isTerminated = true;
+                    break;
+                }
+
+                if (!c.IsHexDigit())
+                {
+                    UngetChar();
+
+                    if (isBracketedSequence)
+                    {
+                        ReportError(_currentIndex, () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    }
+                    else
+                    {
+                        IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                        ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                    }
+
+                    return s_unknownUnicodeChar;
+                }
+
+                sb.Append(c);
+            }
+
+            if (isBracketedSequence && !isTerminated)
+            {
+                c = GetChar();
+                if ((i == maxNumberOfHexDigits) && c.IsHexDigit())
+                {
```

Comment:
Nice!  BTW I did run into a compiler error:
```
error cs0173: type of conditional expression cannot be determined because there is no implicit conversion between 'lambda expression' and 'lambda expression'
```
Fixed with:
```csharp
            if (isBracketedSequence && !isTerminated)
            {
                c = GetChar();
                if (c != '}')
                {
                    UngetChar();
                    ReportError(_currentIndex,
                        i == maxNumberOfHexDigits && c.IsHexDigit()
                            ? (Expression<Func<string>>)(() => ParserStrings.TooManyDigitsInUnicodeEscapeSequence)
                            : () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
                    return s_unknownUnicodeChar;
                }
            }
```
Testing the change now.